src/lib/viewers/doc/DocPreloader.js

```diff
@@ -4,28 +4,26 @@ import {
     CLASS_BOX_PREVIEW_PRELOAD,
     CLASS_BOX_PREVIEW_PRELOAD_CONTENT,
     CLASS_BOX_PREVIEW_PRELOAD_OVERLAY,
-    CLASS_BOX_PREVIEW_PRELOAD_SPINNER,
+    CLASS_BOX_PREVIEW_PRELOAD_PLACEHOLDER,
     CLASS_BOX_PREVIEW_PRELOAD_WRAPPER_DOCUMENT,
     CLASS_INVISIBLE,
     CLASS_IS_TRANSPARENT,
     CLASS_PREVIEW_LOADED,
     PDFJS_CSS_UNITS,
+    PDFJS_HEIGHT_PADDING_PX,
     PDFJS_MAX_AUTO_SCALE,
     PDFJS_WIDTH_PADDING_PX,
-    PDFJS_HEIGHT_PADDING_PX,
 } from '../../constants';
 import { setDimensions, handleRepresentationBlobFetch } from '../../util';
 
 const EXIF_COMMENT_TAG_NAME = 'UserComment'; // Read EXIF data from 'UserComment' tag
 const EXIF_COMMENT_REGEX = /pdfWidth:([0-9.]+)pts,pdfHeight:([0-9.]+)pts,numPages:([0-9]+)/;
 
 const NUM_PAGES_DEFAULT = 2; // Default to 2 pages for preload if true number of pages cannot be read
-const NUM_PAGES_MAX = 500; // Don't show more than 500 placeholder pages
+const NUM_PAGES_MAX = 20;
```

Comment:
500 seems like a lot, but are there any concerns with dropping it down to 20?

Reply:
It will limit how much users can scroll down before the document fully loads. However, when the document loads, the scroll position is reset to the top anyway. It doesn't seem meaningful to have it set to such a high number.